encrypted-media/util/drm-messagehandler.js

```diff
@@ -0,0 +1,230 @@
+                return null;
+            }
+        },
+        'com.microsoft.playready': {
+            responseType: 'arraybuffer',
+            getLicenseMessage: function(response) {
+                return response;
+            },
+            getErrorResponse: function(response) {
+                return String.fromCharCode.apply(null, new Uint16Array(response));
+            },
+            getLicenseRequestFromMessage: function(message) {
+                var msg,
+                    xmlDoc;
+                var licenseRequest = null;
+                var parser = new DOMParser();
+                var dataview = new Uint16Array(message);
+
+                msg = String.fromCharCode.apply(null, dataview);
+                xmlDoc = parser.parseFromString(msg, 'application/xml');
+
+                if (xmlDoc.getElementsByTagName('Challenge')[0]) {
+                    var Challenge = xmlDoc.getElementsByTagName('Challenge')[0].childNodes[0].nodeValue;
+                    if (Challenge) {
+                        licenseRequest = atob(Challenge);
+                    }
+                }
+                return licenseRequest;
+            },
+            getRequestHeadersFromMessage: function(message) {
+                var msg,
+                    xmlDoc;
+                var headers = {};
+                var parser = new DOMParser();
+                var dataview = new Uint16Array(message);
+
+                msg = String.fromCharCode.apply(null, dataview);
+                xmlDoc = parser.parseFromString(msg, 'application/xml');
+
+                var headerNameList = xmlDoc.getElementsByTagName('name');
+                var headerValueList = xmlDoc.getElementsByTagName('value');
+                for (var i = 0; i < headerNameList.length; i++) {
+                    headers[headerNameList[i].childNodes[0].nodeValue] = headerValueList[i].childNodes[0].nodeValue;
+                }
+                // some versions of the PlayReady CDM return 'Content' instead of 'Content-Type'.
+                // this is NOT w3c conform and license servers may reject the request!
+                // -> rename it to proper w3c definition!
+                if (headers.hasOwnProperty('Content')) {
+                    headers['Content-Type'] = headers.Content;
+                    delete headers.Content;
+                }
+                return headers;
+            }
+        }
+    };
+
+    const serverTypes = {
+        'drmtoday': {
+            constructLicenseRequestUrl : function( serverURL, sessionType, messageType, content ) {
+                return serverURL;
+            },
+            getCustomHeaders : function( drmconfig, sessionType ) {
+                var customToken = { outputProtection: { digital : false, analogue: false, enforce: false },
+                                    profile: { purchase: { } },
+                                    storeLicense: ( sessionType === 'persistent-license' ) };
```

Comment:
What are you referring to as "the implementation" here ? Both the license servers expect the type of license and other properties to be in the request. The application needs to provide this information in the appropriate way for each type of server.